content/courses/native-onchain-development/paging-ordering-filtering-data-frontend.md

```diff
@@ -108,11 +108,26 @@ const accounts = await connection.getProgramAccounts(programId, {
 });
 
 accounts.sort((a, b) => {
-  const lengthA = a.account.data.readUInt32LE(0);
-  const lengthB = b.account.data.readUInt32LE(0);
-  const dataA = a.account.data.slice(4, 4 + lengthA);
-  const dataB = b.account.data.slice(4, 4 + lengthB);
-  return dataA.compare(dataB);
+  try {
+    // Check if buffers are long enough to avoid out-of-bounds access
+    const lengthA = a.account.data.readUInt32LE(0);
+    const lengthB = b.account.data.readUInt32LE(0);
+
+    if (
+      a.account.data.length < 4 + lengthA ||
```

Comment:
can you add `const SOME_GOOD_NAME = 4;` per CONTRIBUTING?

I think this is 

```
// 4 bytes are used to store the current length of a string
const STRING_LENGTH_SPACE = 4;
```

Reply:
> can you add `const SOME_GOOD_NAME = 4;` per CONTRIBUTING?
> 
> I think this is
> 
> ```
> // 4 bytes are used to store the current length of a string
> const STRING_LENGTH_SPACE = 4;
> ```

fixed!